src/Native/Unix/gen-buildsys-clang.sh

```diff
@@ -3,6 +3,19 @@
 # This file invokes cmake and generates the build system for Clang.
 #
 
+source="${BASH_SOURCE[0]}"
+
+# resolve $SOURCE until the file is no longer a symlink
+while [[ -h $source ]]; do
+  scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
+  source="$(readlink "$source")"
+
+  # if $source was a relative symlink, we need to resolve it relative to the path where the
+  # symlink file was located
+  [[ $source != /* ]] && source="$scriptroot/$source"
+done
+scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
```

Comment:
Just FYI: in CoreCLR master branch, these scripts are renamed to `find-clang` and `find-gcc`. The code is quite simplified that we do not need lines that require `scriptRoot`. Even before that refactoring took place (e.g. release/3.0 branch), we ended up passing `script_dir` from the caller script to avoid duplicating this logic.

Reply:
Ok thanks for letting know. I don't want to depend on a passed in variable for the scriptroot hence I'll leave it as is for now.